### DISCUSSION TITLE
Remove border for specific window

### DISCUSSION THREAD

**Discussion by gruvw** [qtile/qtile#3850] [Q&A]
I use the Albert launcher (https://github.com/albertlauncher/albert) but I don't want the pop-up search window to have a border.  
How can I specify the border width of a specific application please ?

**Comment by ervinpopescu** [comment]
I don't think you can do that... The only thing you can do is to "turn off" the border width for **all** floating windows.

**Reply by gruvw** [reply]
Ok... this is unfortunate...

**Reply by m-col** [reply]
This kind of configuration of the floating layout has been requested a few times IIRC. Perhaps the `Floating` class could accept some kind of `border_rules` to match border colors and/or sizes to windows. If anybody would be interested in adding this I think it would be well-received, but indeed this is not currently possible.

**Comment by WieeRd** [comment]
![image](https://github.com/user-attachments/assets/daa2f39a-9cd3-4b40-8eb7-255e2a35adc7)

It is very awkward when it happens. It would be great if we can disable borders for the shaped / transparent floating windows.

**Reply by elParaguayo** [reply]
This may be of interest: https://qtile-extras.readthedocs.io/en/stable/manual/ref/borders.html#conditionalborderwidth

**Reply by WieeRd** [reply]
Awesome. I didn't know about `qtile-extra`, looks most promising. But we don't have a way to know if a window is shaped (non-square) or transparent, don't we? Can't really get much out of conditional borders if I am unable to detect that condition :(

**Reply by elParaguayo** [reply]
Which app is this?

**Reply by WieeRd** [reply]
That is Firefox and a notification window about its features that popped up on fresh installation. It no longer pops up now, and internal popups such as context menus does not have those borders. Pretty sure it's because they are internal windows and not separate OS windows. But that makes me wonder why that specific window had borders...

**Reply by WieeRd** [reply]
Should I try re-installing Firefox to get that notification and check the xprop things?

**Reply by elParaguayo** [reply]
I wondered if it was firefox. I don't see that window either.

I'm assuming qtile is responsible for that border so, yes, it would need an xprop to see why we're not ignoring it (I wonder if it's related to https://github.com/qtile/qtile/issues/4662)
